code/game/machinery/computer/atmos_control.dm

```diff
@@ -159,6 +159,25 @@ var/global/list/atmos_controllers = list()
 		alarm.apply_preset(no_cycle_after, 1) //optionally cycle, propagate
 		done_areas += alarm_area
 
+/obj/machinery/computer/atmoscontrol/proc/mass_set_mode(var/mode)
+	if(!mode)
+		return
+
+	var/list/done_areas = list() //a little optimization to avoid needlessly repeating apply_mode()
```

Comment:
You're already avoiding running `apply_mode()` if the current mode is the same as the one you're trying to give it, so this is just extra work for no gain, I think.

Reply:
areas can have multiple air alarms

Reply:
I see, that makes sense.